Sources/Datadog/Datadog.swift

```diff
@@ -134,6 +134,11 @@ public class Datadog {
         instance?.consentProvider.changeConsent(to: trackingConsent)
     }
 
+    /// Clears all data that has not already been sent to Datadog servers.
```

Comment:
maybe it might worth mentioning that this method is async and the files are not deleted once it returns.?

Reply:
Good question 👌. Actually, initially  I made this API `sync` to guarantee "no data" after it returns (by sacrificing the performance ofc), but then I realised that it makes no difference from the user standpoint. The SDK synchronizes all data collection internally so any event collected before `clearAllData()` will be deleted and any event collected after will be stored. That said, adding the information on `async` execution would be purely informative and could be considered a breaking change if we need to use `sync` for some reason in the future. WDYT @buranmert @maxep ?

Reply:
Good question indeed 🤔  Hypothetically, if I do:
```
log(something) // <- will be deleted
clearAllData()
log(something) // <- will be stored
```
Because we are using serial `readWriteQueue` queues in each feature, am I correct?
If so, I don't think we have to specify it's async, the `clearAllData` will be doing what it's expected to do in a optimised manner.